Rimea Tests Update + Reportscript.py + runtest_execute.py

GuidoBasten opened the pull request:
I updated all the Rimea test, all of them exit with success now.
Also, updated the Reportskript.py witch creates a pdf with some important data from the tests.

gjaeger commented:
@GuidoBasten Do we need the pyc-files?

GuidoBasten commented:
No sorry, forgot to delete them and add the whole directory to the repository.

gjaeger commented:
I would like to take a closer look at the pull request. I will not have any capacity until next week.

Ozaq commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -289,10 +319,49 @@ def get_log(testnumber):
+		d_path='_'
+		title='_'
+	os.chdir(start_direc)
+	return(d_path,arc,title)
```
This function looks not like it is doing what it is supposed to be doing based on its name.
From the name I infer that it should return a collection of paths to all diagrams generated for a test based on the tests number. Looking at the code however it does something else:
My understanding is that the functions returns a tuple of: The absolute path of the first PNG file, a bool indicating if any PNG file was found, and a tile string (the file stem) or an underscore if no file was found.

Now I think this can be rewritten very nicely in s few steps. First a small cleanup w.r.t to path handling:
```python
start_direc = os.getcwd()
direc = start_direc + '/test_' + str(testnumber)
os.chdir(direc)
data_list = os.listdir(direc)
```
becomes
```python
test_directory = os.path.join(os.getcwd(), "test_{}".format(testnumer))
```
Note: no change dir

the while loop can now be rewritten as:
```python
diagrams=[]
for filename in os.listdir(directory):
    if filename.endswith(".png"): 
         diagrams.append((os.path.splitext(filename)[0], filename ))
return diagrams
```
or as list expression
```python
return [(os.path.splitext(f)[0], f ) for f in os.listdir(directory) if f.endswith(".png")]
```

Ozaq commented on Utest/rimea_tests/test_1/runtest_rimea_1.py:
```diff
@@ -48,8 +50,11 @@ def run_rimea_test1(inifile, trajfile):
 
+
 if __name__ == "__main__":
+    start_time=time.time()
```
Instead of repeating the same code over and over in each test file, isn't there a way to make JPSRunTestDriver emit the output you need?

GuidoBasten replied:
Year this would be nicer, but if we change this we also have to change the python scrips for the Juelich test and all other test witch are working witch the JPSRunTestDriver.
So i think this will be a little Project for the future but now  I just want to get Rime running.

xuqiancheng commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -245,13 +258,17 @@ def get_git_status():
 		"""
 		## Get the branch name by `git status` command
-		branch = subprocess.check_output(["git", "status"]).splitlines()[0].split(' ')[-1]
+		branch =str(subprocess.check_output(["git", "status"]))
```
This code doesn't work on windows. Is there any way to solve it?

chraibi replied:
Why it does not work?
Can you post the error?

xuqiancheng replied:
> Why it does not work?
> Can you post the error?

I think we can not call git on windows in this way
![image](https://user-images.githubusercontent.com/32732507/62128926-d0a89c80-b2d5-11e9-8f51-388f9996f6f2.png)

gjaeger replied:
@xuqiancheng Any suggestions?

Ozaq replied:
IIRC the windows installer for git does not add git to the PATH variable automatically. When you start the git shell the environment gets populated with the correct path. For any other program to find git on windows the path to git has to be added to PATH

See:https://stackoverflow.com/questions/26620312/installing-git-in-path-with-github-client-for-windows

chraibi replied:
Maybe you can try this hack first: 

```
import subprocess

proc = subprocess.Popen('git.cmd status')
print ('result: ', proc.communicate())
```

https://stackoverflow.com/questions/4965175/make-subprocess-find-git-executable-on-windows

Ozaq replied:
> Maybe you can try this hack first:
> 
> ```
> import subprocess
> 
> proc = subprocess.Popen('git.cmd status')
> print ('result: ', proc.communicate())
> ```
> 
> https://stackoverflow.com/questions/4965175/make-subprocess-find-git-executable-on-windows

Ugh, so the solution would be to search for both `git.exe` or `git.cmd` on windows and then use the respective call :/

xuqiancheng replied:
@gjaeger I  have tried several ways from internet, but didn't work.

@Ozaq Thanks, but I have already added git to the PATH variable, and I can use git in cmd
![image](https://user-images.githubusercontent.com/32732507/62132097-b32b0100-b2dc-11e9-9121-640971cd3220.png)

@chraibi I have tried several ways including this one, but didn't work.
![image](https://user-images.githubusercontent.com/32732507/62132266-0309c800-b2dd-11e9-8efb-d54000d224c4.png)

xuqiancheng replied:
@Ozaq 
You are right, I find that I have made some mistakes with the path of git and pycharm. 
After adjusting, it works good now. 
It's my fault, the code works on windows.

Ozaq replied:
> @Ozaq
> You are right, I find that I have made some mistakes with the path of git and pycharm.
> After adjusting, it works good now.
> It's my fault, the code works on windows.

Nice! No extra code for windows! Yay!

xuqiancheng commented on Utest/rimea_tests/test_4/runtest_rimea_4.py:
```diff
@@ -54,13 +54,15 @@
-    density = [2.0,3.0,4.0,0.5,1.0,1.5] #DO NOT CHANGE THE ORDER
-    
+#    density = [2.0,3.0,4.0,0.5,1.0,1.5] #DO NOT CHANGE THE ORDER
+    density =  [2.0,1.0,6.0,3.0,4.0,5.0,0.5]
```
I think should be "density=[2, 3, 4, 0.5, 5, 6, 1]", according to the name of trajecory files

xuqiancheng replied:
> I think should be "density=[2, 3, 4, 0.5, 5, 6, 1]", according to the name of trajecory files
The order in windows is like this.May be you can sort trajectories to avoid it.

![image](https://user-images.githubusercontent.com/32732507/62083632-b3cd8400-b257-11e9-8dce-4a5e8f1ba56e.png)

gjaeger replied:
@GuidoBasten Why do not you sort in ascending order?

chraibi replied:
@GuidoBasten this is actually a good question.

GuidoBasten commented:
@gjaeger the png file is the logo from JuPedSim, witch we need for the first page of the report script

@Ozaq I fixed the issue with the years at line 21 as you suggested and changed the get_diagrams function.

@xuqiancheng I took a look at the densities for test_4. Python passes the trajectory's in a other order as i expected. At the moment i don't know why it uses this order but the diagram seems to make sense by using it. In the attachment is a Picture of the Order.
I don't actually have a  Windows os were JuPedSim is installed. Does it put any error message out?
I will try to make a request witch os is used.

I will take care of this and push my new changes soon.
![Rimea_4](https://user-images.githubusercontent.com/51946510/62079650-736a0800-b24f-11e9-8a4e-379856efa072.png)

gjaeger commented:
@GuidoBasten 

- `Utest/rimea_tests/test_13/flow.png`
- Utest/rimea_tests/test_4/fundamental_diag.png`
- `Utest/rimea_tests/test_4/fundamental_diag.txt`
- `Utest/rimea_tests/test_7/ids_velocities.csv`
- `Utest/rimea_tests/test_7/velocity_distribution.png`

What about these files? Do we need these or are they created?

gjaeger commented on Utest/rimea_tests/test_1/master_ini.xml:
```diff
@@ -10,7 +10,7 @@
-	<trajectories format="xml-plain" fps="8">
+	<trajectories format="xml-plain" fps="8" color_mode="velocity">
 		<file location="RiMEATest1_traj.xml" />
 		<!--<socket hostname="127.0.0.1" port="8989"/> -->
```
Do we need these comments?

chraibi commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -268,14 +285,27 @@ def get_tests_status(testnumber):
+                                        					line):
+                                        					status = "Succeed"
+                                       						break
+
```
Please check the indentation. 8 spaces are too much.
4 spaces are enough.

gjaeger commented on Utest/rimea_tests/test_5/runtest_rimea_5.py:
```diff
@@ -1,94 +1,97 @@
-    logging.info("%s exits with SUCCESS" % (argv[0]))
+#!/usr/bin/env python3
+"""
+Test description
```
Should we include this description in the report?

schroedtert commented:
I ran the test of develop (before this PR). This are the results:
```
      Start  1: test_clean_geometry_1
 1/50 Test  #1: test_clean_geometry_1 ............   Passed    4.53 sec
      Start  2: test_clean_geometry_2
 2/50 Test  #2: test_clean_geometry_2 ............   Passed   25.71 sec
      Start  3: test_clean_geometry_3
 3/50 Test  #3: test_clean_geometry_3 ............Child aborted***Exception:   0.10 sec
      Start  4: test_clean_geometry_4
 4/50 Test  #4: test_clean_geometry_4 ............   Passed    0.98 sec
      Start  5: test_clean_geometry_5
 5/50 Test  #5: test_clean_geometry_5 ............   Passed    0.42 sec
      Start  6: test_clean_geometry_6
 6/50 Test  #6: test_clean_geometry_6 ............   Passed    5.94 sec
      Start  7: test_clean_geometry_7
 7/50 Test  #7: test_clean_geometry_7 ............   Passed  316.56 sec
      Start  8: test_clean_geometry_8
 8/50 Test  #8: test_clean_geometry_8 ............   Passed    0.37 sec
      Start  9: test_clean_geometry_9
 9/50 Test  #9: test_clean_geometry_9 ............   Passed    1.77 sec
      Start 10: test_clean_geometry_10
10/50 Test #10: test_clean_geometry_10 ...........   Passed    3.00 sec
      Start 11: test_clean_geometry_11
11/50 Test #11: test_clean_geometry_11 ...........   Passed    1.32 sec
      Start 12: runtest_dev_1
12/50 Test #12: runtest_dev_1 ....................***Failed   23.65 sec
      Start 13: runtest_juelich_1
13/50 Test #13: runtest_juelich_1 ................   Passed    3.69 sec
      Start 14: runtest_juelich_10
14/50 Test #14: runtest_juelich_10 ...............***Failed    3.07 sec
      Start 15: runtest_juelich_11
15/50 Test #15: runtest_juelich_11 ...............   Passed    3.41 sec
      Start 16: runtest_juelich_12
16/50 Test #16: runtest_juelich_12 ...............   Passed    6.69 sec
      Start 17: runtest_juelich_14
17/50 Test #17: runtest_juelich_14 ...............***Failed    2.61 sec
      Start 18: runtest_juelich_15
18/50 Test #18: runtest_juelich_15 ...............   Passed    1.34 sec
      Start 19: runtest_juelich_2
19/50 Test #19: runtest_juelich_2 ................   Passed    1.85 sec
      Start 20: runtest_juelich_3
20/50 Test #20: runtest_juelich_3 ................   Passed    1.26 sec
      Start 21: runtest_juelich_4
21/50 Test #21: runtest_juelich_4 ................   Passed    7.56 sec
      Start 22: runtest_juelich_5
22/50 Test #22: runtest_juelich_5 ................   Passed    1.73 sec
      Start 23: runtest_juelich_6
23/50 Test #23: runtest_juelich_6 ................   Passed    3.42 sec
      Start 24: runtest_juelich_8
24/50 Test #24: runtest_juelich_8 ................   Passed    5.47 sec
      Start 25: runtest_juelich_9
25/50 Test #25: runtest_juelich_9 ................   Passed   17.24 sec
      Start 26: runtest_execute
26/50 Test #26: runtest_execute ..................   Passed    0.79 sec
      Start 27: runtest_rimea_1
27/50 Test #27: runtest_rimea_1 ..................***Failed    1.93 sec
      Start 28: runtest_rimea_10
28/50 Test #28: runtest_rimea_10 .................   Passed   15.71 sec
      Start 29: runtest_rimea_11
29/50 Test #29: runtest_rimea_11 .................   Passed  132.66 sec
      Start 30: runtest_rimea_12
30/50 Test #30: runtest_rimea_12 .................   Passed  655.40 sec
      Start 31: runtest_rimea_13
31/50 Test #31: runtest_rimea_13 .................***Failed    4.24 sec
      Start 32: runtest_rimea_14
32/50 Test #32: runtest_rimea_14 .................***Failed    2.32 sec
      Start 33: runtest_rimea_15
33/50 Test #33: runtest_rimea_15 .................***Failed    2.96 sec
      Start 34: runtest_rimea_2
34/50 Test #34: runtest_rimea_2 ..................   Passed    3.89 sec
      Start 35: runtest_rimea_3
35/50 Test #35: runtest_rimea_3 ..................   Passed    3.68 sec
      Start 36: runtest_rimea_4
36/50 Test #36: runtest_rimea_4 ..................***Failed  2789.53 sec
      Start 37: runtest_rimea_5
37/50 Test #37: runtest_rimea_5 ..................***Failed    7.73 sec
      Start 38: runtest_rimea_6
38/50 Test #38: runtest_rimea_6 ..................   Passed    6.74 sec
      Start 39: runtest_rimea_7
39/50 Test #39: runtest_rimea_7 ..................***Failed  195.77 sec
      Start 40: runtest_rimea_8
40/50 Test #40: runtest_rimea_8 ..................***Timeout 10800.11 sec
      Start 41: runtest_rimea_9
41/50 Test #41: runtest_rimea_9 ..................***Failed  6225.73 sec
      Start 42: runtest_router_1
42/50 Test #42: runtest_router_1 .................   Passed    1.83 sec
      Start 43: runtest_router_2
43/50 Test #43: runtest_router_2 .................   Passed    0.93 sec
      Start 44: runtest_router_3
44/50 Test #44: runtest_router_3 .................   Passed    0.88 sec
      Start 45: runtest_router_4
45/50 Test #45: runtest_router_4 .................   Passed    0.90 sec
      Start 46: runtest_router_5
46/50 Test #46: runtest_router_5 .................   Passed    0.88 sec
      Start 47: runtest_router_6
47/50 Test #47: runtest_router_6 .................   Passed   78.52 sec
      Start 48: runtest_router_7
48/50 Test #48: runtest_router_7 .................   Passed    1.37 sec
      Start 49: runtest_router_8
49/50 Test #49: runtest_router_8 .................   Passed    0.84 sec
      Start 50: runtest_router_9
50/50 Test #50: runtest_router_9 .................   Passed   26.09 sec

74% tests passed, 13 tests failed out of 50

Total Test time (real) = 21405.61 sec

The following tests FAILED:
	  3 - test_clean_geometry_3 (Child aborted)
	 12 - runtest_dev_1 (Failed)
	 14 - runtest_juelich_10 (Failed)
	 17 - runtest_juelich_14 (Failed)
	 27 - runtest_rimea_1 (Failed)
	 31 - runtest_rimea_13 (Failed)
	 32 - runtest_rimea_14 (Failed)
	 33 - runtest_rimea_15 (Failed)
	 36 - runtest_rimea_4 (Failed)
	 37 - runtest_rimea_5 (Failed)
	 39 - runtest_rimea_7 (Failed)
	 40 - runtest_rimea_8 (Timeout)
	 41 - runtest_rimea_9 (Failed)
Errors while running CTest
```
Maybe worth taking a look at the failed ones and especially at the runtimes of particular ones. E.g. rimea_4 2500s (approx 40 min), rimea_8 10800s (approx 3h + timeout!), rimea_9 3900s (approx 1h). These 3 tests also fail. 

I will update this, with the results of this PR after the test have been executed.

schroedtert commented:
As promised, here the output of the test with this PR:
```
Test project /Users/tobias/development/jupedsim-github/jpscore/build-443
      Start  1: test_clean_geometry_1
 1/50 Test  #1: test_clean_geometry_1 ............   Passed    0.41 sec
      Start  2: test_clean_geometry_2
 2/50 Test  #2: test_clean_geometry_2 ............   Passed   16.13 sec
      Start  3: test_clean_geometry_3
 3/50 Test  #3: test_clean_geometry_3 ............Child aborted***Exception:   0.05 sec
      Start  4: test_clean_geometry_4
 4/50 Test  #4: test_clean_geometry_4 ............   Passed    0.55 sec
      Start  5: test_clean_geometry_5
 5/50 Test  #5: test_clean_geometry_5 ............   Passed    0.20 sec
      Start  6: test_clean_geometry_6
 6/50 Test  #6: test_clean_geometry_6 ............   Passed    3.58 sec
      Start  7: test_clean_geometry_7
 7/50 Test  #7: test_clean_geometry_7 ............   Passed  276.81 sec
      Start  8: test_clean_geometry_8
 8/50 Test  #8: test_clean_geometry_8 ............   Passed    0.35 sec
      Start  9: test_clean_geometry_9
 9/50 Test  #9: test_clean_geometry_9 ............   Passed    1.62 sec
      Start 10: test_clean_geometry_10
10/50 Test #10: test_clean_geometry_10 ...........   Passed    2.54 sec
      Start 11: test_clean_geometry_11
11/50 Test #11: test_clean_geometry_11 ...........   Passed    1.27 sec
      Start 12: runtest_dev_1
12/50 Test #12: runtest_dev_1 ....................***Failed   21.90 sec
      Start 13: runtest_juelich_1
13/50 Test #13: runtest_juelich_1 ................   Passed    2.86 sec
      Start 14: runtest_juelich_10
14/50 Test #14: runtest_juelich_10 ...............***Failed    2.71 sec
      Start 15: runtest_juelich_11
15/50 Test #15: runtest_juelich_11 ...............   Passed    2.52 sec
      Start 16: runtest_juelich_12
16/50 Test #16: runtest_juelich_12 ...............   Passed    5.59 sec
      Start 17: runtest_juelich_14
17/50 Test #17: runtest_juelich_14 ...............***Failed    2.22 sec
      Start 18: runtest_juelich_15
18/50 Test #18: runtest_juelich_15 ...............   Passed    1.07 sec
      Start 19: runtest_juelich_2
19/50 Test #19: runtest_juelich_2 ................   Passed    1.62 sec
      Start 20: runtest_juelich_3
20/50 Test #20: runtest_juelich_3 ................   Passed    1.08 sec
      Start 21: runtest_juelich_4
21/50 Test #21: runtest_juelich_4 ................   Passed    6.43 sec
      Start 22: runtest_juelich_5
22/50 Test #22: runtest_juelich_5 ................   Passed    1.47 sec
      Start 23: runtest_juelich_6
23/50 Test #23: runtest_juelich_6 ................   Passed    2.89 sec
      Start 24: runtest_juelich_8
24/50 Test #24: runtest_juelich_8 ................   Passed    4.65 sec
      Start 25: runtest_juelich_9
25/50 Test #25: runtest_juelich_9 ................   Passed   16.00 sec
      Start 26: runtest_execute
26/50 Test #26: runtest_execute ..................   Passed    0.12 sec
      Start 27: runtest_rimea_1
27/50 Test #27: runtest_rimea_1 ..................   Passed    2.10 sec
      Start 28: runtest_rimea_10
28/50 Test #28: runtest_rimea_10 .................   Passed   14.38 sec
      Start 29: runtest_rimea_11
29/50 Test #29: runtest_rimea_11 .................   Passed    1.25 sec
      Start 30: runtest_rimea_12
30/50 Test #30: runtest_rimea_12 .................   Passed  542.10 sec
      Start 31: runtest_rimea_13
31/50 Test #31: runtest_rimea_13 .................***Failed    2.54 sec
      Start 32: runtest_rimea_14
32/50 Test #32: runtest_rimea_14 .................***Failed  201.91 sec
      Start 33: runtest_rimea_15
33/50 Test #33: runtest_rimea_15 .................***Failed  833.36 sec
      Start 34: runtest_rimea_2
34/50 Test #34: runtest_rimea_2 ..................   Passed    2.05 sec
      Start 35: runtest_rimea_3
35/50 Test #35: runtest_rimea_3 ..................   Passed    1.56 sec
      Start 36: runtest_rimea_4
36/50 Test #36: runtest_rimea_4 ..................   Passed  7167.49 sec
      Start 37: runtest_rimea_5
37/50 Test #37: runtest_rimea_5 ..................   Passed    5.76 sec
      Start 38: runtest_rimea_6
38/50 Test #38: runtest_rimea_6 ..................   Passed    5.76 sec
      Start 39: runtest_rimea_7
39/50 Test #39: runtest_rimea_7 ..................   Passed  174.99 sec
      Start 40: runtest_rimea_8
40/50 Test #40: runtest_rimea_8 ..................   Passed  4555.44 sec
      Start 41: runtest_rimea_9
41/50 Test #41: runtest_rimea_9 ..................***Timeout 10799.87 sec
      Start 42: runtest_router_1
42/50 Test #42: runtest_router_1 .................   Passed    1.80 sec
      Start 43: runtest_router_2
43/50 Test #43: runtest_router_2 .................   Passed    1.21 sec
      Start 44: runtest_router_3
44/50 Test #44: runtest_router_3 .................   Passed    1.17 sec
      Start 45: runtest_router_4
45/50 Test #45: runtest_router_4 .................   Passed    1.15 sec
      Start 46: runtest_router_5
46/50 Test #46: runtest_router_5 .................   Passed    1.11 sec
      Start 47: runtest_router_6
47/50 Test #47: runtest_router_6 .................   Passed   91.62 sec
      Start 48: runtest_router_7
48/50 Test #48: runtest_router_7 .................   Passed    1.51 sec
      Start 49: runtest_router_8
49/50 Test #49: runtest_router_8 .................   Passed    1.11 sec
      Start 50: runtest_router_9
50/50 Test #50: runtest_router_9 .................   Passed   30.78 sec

84% tests passed, 8 tests failed out of 50

Total Test time (real) = 24819.06 sec

The following tests FAILED:
	  3 - test_clean_geometry_3 (Child aborted)
	 12 - runtest_dev_1 (Failed)
	 14 - runtest_juelich_10 (Failed)
	 17 - runtest_juelich_14 (Failed)
	 31 - runtest_rimea_13 (Failed)
	 32 - runtest_rimea_14 (Failed)
	 33 - runtest_rimea_15 (Failed)
	 41 - runtest_rimea_9 (Timeout)
Errors while running CTest
make: *** [test] Error 8
```
Totol test time is almost 7h!, which is way too long in my opinion. At least for the test shipped to the users

GuidoBasten commented:
@schroedtert  saw the Problem with test 14 and 15.
The name name of the flow exit files changed again.
I changed it and test 14 work again.

Test 15 also works but it doesn't exit with success anymore cause the H-lines don't work.
Last time i tried this Test (2 weeks ago) they were still Working.
Do we change something by the H-lines ?
Maybe just the way to create the geometry routing file, so this is the one witch worked 2 weeks ago, but not anymore

![H_lines_rimea_15](https://user-images.githubusercontent.com/51946510/62198098-4e75b200-b381-11e9-9802-36ec54f76274.png)

I'll also take a look at test 9, last time i tried this one i took 3943.744 seconds.
In this Test we are also need flow exit files, so i am a little bit confused why it doesn't fail

GuidoBasten commented:
So I changed the comments that open the flow exit files and every test now works in an acceptable time.
Test 15 is the only one witch doesn't exit with success, but it also executes.


![Rimea](https://user-images.githubusercontent.com/51946510/62410272-e5da3f80-b5e3-11e9-9204-d9efb991ab89.png)


The last step I'll do is to check witch comments can be delete and maybe add some test descriptions to the Report-script

schroedtert commented on .gitlab-ci.yml:
```diff
@@ -13,9 +13,10 @@ stages:
   - compile
   - unit_test
-  - dev_test
   # - juelich_tests
```
@chraibi is the .gitlab-ci still needed or should we remove it as we are no longer working on gitlab?

chraibi replied:
@schroedtert I would remove it.

Ozaq commented on Utest/rimea_tests/test_15/runtest_rimea_15.py:
```diff
@@ -117,6 +104,7 @@ def create_dia (peds):
     logging.info("%s exits with SUCCESS\nExecution time %.3f seconds." % (argv[0],time.time()-start_time))
     exit(SUCCESS)
+
+create_dia(500)
```
This looks wrong. You have 
```python
if __name__ == "__main__":
    ...
    exit(SUCCESS)
```
so that line `create_dia(500)` will only be called if this file is imported instead of executed. Is this intended?

GuidoBasten replied:
No it isn't. I took it back up i the if-statement.

Ozaq commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -1,348 +1,311 @@
+    :return: null
+    """
 
+    # Define titlepage in title environment
```
Indentation still 8 instead of 4 ws.

Ozaq commented:
@GuidoBasten I looked into the build failure on Travis, its a Travis network connection Problem. Looks like they had a short internal outage to their apt package cache and boost could not be installed.

GuidoBasten commented:
So this is how the Report script looks like (with the added descriptions)

gjaeger commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -1,30 +1,28 @@
-	:return: All evacuation times(list)
-	"""
-	log_name = "./test_" + str(testnumber) + "/log_test_" + str(testnumber) + ".txt"
+    """
```
What's the difference?

GuidoBasten replied:
Didn't add the last changes to my repository, so in that file the comments are still there. 
Sorry I will take care of this next time.

gjaeger commented on Utest/rimea_tests/reportscript.py:
```diff
@@ -296,15 +265,19 @@ def get_log(testnumber):
-    diagrams = []
+    is_diagram = False
 
     for filename in os.listdir(test_directory):
```
If we take the images from the RiMEA document, we must cite the source.

gjaeger commented:
@GuidoBasten The pdf-file is generated from the script `Utest/rimea_tests/reportscript.py`?

GuidoBasten commented:
Yes it is, but with old logs and diagrams. Just wanted to test the reportscript.py, so i used this files from an older version cause i don't want to run all tests again. In test 4 I didn't even use a log, just create a Text file with the needed Name.

GuidoBasten commented:
Just saw the .tex data for the test descriptions are also not in the repository.
Tried to add them but it doesn't work.
I don't know why, it just ignores them.
I will try it later with another pc, maybe this works.

gjaeger commented:
@GuidoBasten 
_.tex_-files are ignored. 

See `man git-add`: 
```
   -f, --force
       Allow adding otherwise ignored files.
```

So, you can force it: `git add --force file.foo`

chraibi commented:
@GuidoBasten how are the chances to finish this soon?

GuidoBasten commented:
So I think we could merge the Project now, or is there anything else you need me to change before?

zeroset commented:
What is about all the png files? Are they generated or do we need them as binary files in the repository?

GuidoBasten commented:
The Diagrams in the test folders will be created by running the test. With the exception of the stairs.png in test one/two and the Distribution speed png from test 13. I just saw that i didn't delete the parameter.png from test_8. All the other png files are neded to created the Test descriptions and the reportscript.pfd. These will not be created by running the tests.
